### PR TITLE
Update RAJA version constraints and expose OMP task option

### PR DESCRIPTION
**Pull request by rhornung67**
Updates to RAJA Spack package after latest release + expose option for needed control in GitLab CI testing

### CODE REVIEW

**Review by MrBurmark — APPROVED** [review]

File: packages/raja/package.py [hunk 1]
```diff
@@ -191,6 +194,8 @@ def initconfig_package_entries(self):
 
         entries.append(cmake_cache_option("RAJA_ENABLE_VECTORIZATION", "+vectorization" in spec))
 
+        entries.append(cmake_cache_option("RAJA_ENABLE_OPENMP_TASK", "+omptask" in spec))
```
Comment (bmhan12) [inline packages/raja/package.py:197]: Does `omptask` need to be added as a variant?

Comment (rhornung67) [reply packages/raja/package.py:197]: D'oh! Missed that. Thank you.

**Comment by adrienbernede** [comment]
@rhornung67 I will have a further look into this and approve asap. I want to see the test reliably passing.

Also, we appear to have the issue only with clang compilers. So would you agree that we will use this variant on only when sing clang in CI ?

You said you don't use that in RAJA, what did you mean ? I wouldn't want this to become the reason for untested code.

**Comment by rhornung67** [comment]
When I get things up and running on RAJA and RAJAPerf, I will adjust the specs we use to turn on OpenMP task for compilers and versions that it works for. We will fall back to another implementation for the others.

The issue only appears with clang13 and clang14 on lassen with the updated CI stuff. These builds and tests were passing on the develop branch. This is the part that bothers me, because I cannot reproduce the failures outside of GitLab CI, even when I run the build and test script with the Spack specs we use in CI.

The RAJAPerf kernel variants that seg fault have no RAJA code in them. That is, they are raw OpenMP, we call them base variants that we use for comparison.

Now I'm running into issues on azure pipelines running out of disk space. When it rains, it pours.

**Comment by rhornung67** [comment]
@adrienbernede hold off on merging this. I need to make a few more changes.

**Review by adrienbernede — APPROVED** [review]

**Pull request merged**